components/admin_console/admin_definition.jsx

```diff
@@ -4124,3 +4125,28 @@ t('admin.field_names.postEditTimeLimit');
 t('admin.field_names.restrictCreationToDomains');
 t('admin.field_names.restrictDirectMessage');
 t('admin.field_names.teammateNameDisplay');
+
+let cache = null;
+const adminDefinitionsPlugins = {};
+
+export function registerAdminConsolePlugin(pluginId, func) {
+    adminDefinitionsPlugins[pluginId] = func;
+    cache = null;
+}
+
+export function unregisterAdminConsolePlugin(pluginId) {
+    delete adminDefinitionsPlugins[pluginId];
+    cache = null;
+}
+
+export default function getAdminDefinition() {
+    if (cache === null) {
```

Comment:
I cache this and only regenerate it the first time or when a plugin registers an admin console plugin, or when when this kind of plugin is deleted.